### ISSUE TITLE
File Upload

### ISSUE THREAD

**Issue by olzii** [kriasoft/nodejs-api-starter#216]
Do anyone have plan to add file upload feature ?

**Comment by koistya** [comment]
Yep, will add it. It would look like an API endpoint that generates upload URL (to Google Storage Bucket or Amazon S3). The client app would request that URL and upload file(s) directly to cloud storage using `fetch({ method: "PUT", ... })`.

```graphql
mutation {
  createUploadUrl(...)
}
```

**Comment by phpb-com** [comment]
Any update on this? Thanks.

**Comment by koistya** [comment]
@phpb-com yes, there is `api/src/mutations/getUploadURL.ts` and `api/src/mutations/saveUpload.ts` on the API side. All you have to do on the client is to request "upload URL", send the file(s) via HTTP PUT to that URL, then send "uploaded file URL" to the `saveUpload()` API mutation (which copies the file over from a temporary upload bucket location to a permanent place based on the linked content type (user photo, blog post pictures, etc.). As was described [here](https://github.com/kriasoft/relay-starter-kit/issues/216#issuecomment-665617784).
